From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mm: remove folio_test_transhuge()
Date: Wed, 16 Aug 2023 16:11:59 +0100

This function is misleading; people think it means "Is this a THP", when
all it actually does is check whether this is a large folio.  Remove it;
the one remaining user should have been checking to see whether the folio
is PMD sized or not.

Link: https://lkml.kernel.org/r/20230816151201.3655946-12-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Cc: Yanteng Si <siyanteng@loongson.cn>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/page-flags.h |    5 -----
 mm/memcontrol.c            |    2 +-
 2 files changed, 1 insertion(+), 6 deletions(-)

--- a/include/linux/page-flags.h~mm-remove-folio_test_transhuge
+++ a/include/linux/page-flags.h
@@ -853,11 +853,6 @@ static inline int PageTransHuge(struct p
 	return PageHead(page);
 }
 
-static inline bool folio_test_transhuge(struct folio *folio)
-{
-	return folio_test_head(folio);
-}
-
 /*
  * PageTransCompound returns true for both transparent huge pages
  * and hugetlbfs pages, so it should only be called when it's known
--- a/mm/memcontrol.c~mm-remove-folio_test_transhuge
+++ a/mm/memcontrol.c
@@ -5755,7 +5755,7 @@ static int mem_cgroup_move_account(struc
 		if (folio_mapped(folio)) {
 			__mod_lruvec_state(from_vec, NR_ANON_MAPPED, -nr_pages);
 			__mod_lruvec_state(to_vec, NR_ANON_MAPPED, nr_pages);
-			if (folio_test_transhuge(folio)) {
+			if (folio_test_pmd_mappable(folio)) {
 				__mod_lruvec_state(from_vec, NR_ANON_THPS,
 						   -nr_pages);
 				__mod_lruvec_state(to_vec, NR_ANON_THPS,
_
